tests/search/query_profiling/query_profiling.rb

```diff
@@ -0,0 +1,157 @@
+# Copyright Yahoo. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+
+require 'indexed_search_test'
+require 'json'
+
+class QueryProfiling < IndexedSearchTest
+  
+  def setup
+    @match_tag = "match_profiling"
+    @first_phase_tag = "first_phase_profiling"
+    @second_phase_tag = "second_phase_profiling"
+    set_owner('havardpe')
+    set_description("Test query profiling (matching/ranking)")
+  end
+
+  def test_rank_profiling
+    deploy_app(SearchApp.new.sd(selfdir + 'test.sd'))
+    start
+    feed_and_wait_for_docs('test', 5, :file => selfdir + 'docs.json')
+    for depth in [1,3,-1,-3] do
+      verify_match_profiling(depth)
+      verify_first_phase_profiling(depth)
+      verify_second_phase_profiling(depth)
+    end
+  end
+
+  def get_query_trace(query)
+    result = search(query).json
+    trace = result["trace"]["children"][1]["children"][0]["children"][1]["message"][0]["traces"][0]
+    assert_equal(trace["tag"], "query_execution")
+    return trace
+  end
+  
+  def verify_match_profiling(depth)
+    puts "test case: match(#{depth})"
+    verify_match_trace(get_query_trace("/search/?query=sddocname:test&format=json&hits=1&tracelevel=1&trace.profileDepth=#{depth}&type=all"),
```

Comment:
Consider adding a function that builds the query string, avoiding duplication of the parts that are not relevant for the test cases. Now I had to scan all query strings (looking for the actual differences), making the test cases harder to read. Also, the "/search/" prefix should not be needed when calling search(query).